### ISSUE TITLE
Links are not crawlable with acceptable URL

### ISSUE THREAD

**Issue by JPPhoto** [GoogleChrome/lighthouse#11818]
#### Provide the steps to reproduce
1. Run LH on https://www.cupcakeproject.com/best-vanilla-cupcake-recipe/
  

#### What is the current behavior?
I see "Links are not crawlable" for this element:

```
<a href="https://www.pinterest.com/pin/create/bookmarklet/?url=https%3A%2F%2Fwww.cu…" onclick="window.open(this.href,'targetWindow','toolbar=no,location=no,status=no,men…" data-recipe="21315" style="color: #333333;" class="wprm-recipe-pin wprm-recipe-link wprm-block-text-normal" target="_blank" rel="nofollow noopener noopener noreferrer">
```
#### What is the expected behavior?
Since the full link is crawlable, I wouldn't expect to see this error. The link has urlencoded parts, so perhaps that's triggering an issue; opening the URL in a browser shows the expected page.

#### Environment Information
* Affected Channels: All
* Lighthouse version: 6.3.0
* Chrome version: 87.0.4280.88
* Node.js version:
* Operating System: Windows 10 20H2

**Related issues**

**Comment by patrickhulce** [comment]
Thanks for filing @JPPhoto!

I'm really surprised this case of a valid `href` + `onclick=window.open` wasn't explicitly covered in https://github.com/GoogleChrome/lighthouse/issues/10590 or anywhere in our tests 😐 Conceptually I think this should be fine for the same reasons as https://github.com/GoogleChrome/lighthouse/issues/11443#issuecomment-694898412.

cc @AVGP in case this should still be considered failing for a reason I'm not aware of :)

**Comment by JPPhoto** [comment]
@patrickhulce I should add that the URI is valid, whether or not the onclick JS is invoked. I get that Lighthouse would report an error _iff_ the onclick handler tried to deceive the user; perhaps this is rather hard to detect and it's easier to flag instances as errors? Having not looked at the code, I don't know what's going on under the hood.

**Comment by connorjclark** [comment]
1) we'd have to click every link and make sure the accompanying network request actually matches the provided href value... 

2) we'd have to make sure that search/crawlers even try to resolve links with an onclick... or just ignore onclick and use href ? cc @AVGP any thoughts here?

ideally we could just check for href and ignore all else ... but only if crawlers actually work like that :)

**Comment by AVGP** [comment]
So as far as Google Search is concerned, look at the `href` and that's it. Googlebot only has an issue with hrefs like `javascript:doStuff('home')` or `#` as none of these are crawlable URLs (well the empty fragment is, but it's just the current page anyway)

**Comment by patrickhulce** [comment]
thanks @AVGP! We'll mark a valid href + onclick handler as passing then 👍

**Comment by svkrclg** [comment]
I can pick this up @patrickhulce. Upon looking at the codebase, [this](https://github.com/GoogleChrome/lighthouse/blob/e58aa09a29c97d93ed039092093f4fb3351c9155/lighthouse-core/audits/seo/crawlable-anchors.js#L65) is where the anchor is getting marked as failed.
Few doubts here:
- why have we marked `window.open` as failed anchor tag?
- How can we verify `onclick` handler?

**Comment by patrickhulce** [comment]
> why have we marked window.open as failed anchor tag?

The purpose of this audit is to flag links that are not crawlable. `<a onclick="window.open('http://example.com')">` is not crawlable.

> How can we verify onclick handler as valid?

We aren't. This change is about considering an anchor valid if it has a valid `href`, regardless of onclick behavior.

**Comment by paulirish** [comment]
fixed by #13105. shipped in LH 8.6.0
